front/src/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,622 @@
+            user_name,
+            email,
+            mobile,
+            id,
+            role,
+            password,
+          }));
+          setUserName(user_name || "");
+          setDisplayName(display_name || "");
+          setEmail(email || "");
+          setMobile(mobile || "");
+          setRole(role || "");
+          setUserId(id);
+          setPassword(password || "");
+        }
+      } catch (error) {
+        console.error("Error fetching user details:", error);
+      }
+    };
+    fetchUserDetails();
+  }, []);
+
+  useEffect(() => {
+    if (userDetails) {
+      setUserName(userDetails.user_name || "");
+      setDisplayName(userDetails.display_name || "");
+      setEmail(userDetails.email || "");
+      setMobile(userDetails.mobile || "");
+      setRole(userDetails.role || "");
+    }
+  }, [userDetails]);
+
+  useEffect(() => {
+    getApplications();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getApplications = async () => {
+    const session = await getSession();
+    try {
+      if (session) {
+        const { user_name, email, mobile, role, password } = session.user;
+        setUserName(user_name || "");
+        setDisplayName(display_name || "");
+        setMobile(mobile || "");
+        setEmail(email || "");
+        setRole(role || "");
+        setPassword(password || "");
+      }
+    } catch (error: any) {
+      console.log(error);
+    }
+  };
+
+  const handlePasswordVisibility = (field: number) => {
+    switch (field) {
+      case 1:
+        setShowPassword1((prevShowPassword) => !prevShowPassword);
+        break;
+      case 2:
+        setShowPassword2((prevShowPassword) => !prevShowPassword);
+        break;
+      case 3:
+        setShowPassword3((prevShowPassword) => !prevShowPassword);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setEditedData((prevData) => ({ ...prevData, [name]: value }));
+    if (name === "user_name") {
+      setUserName(value);
+    } else if (name === "display_name") {
+      setDisplayName(value);
+    }
+  };
+
+  const validateEmail = (email: string): boolean => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const validateProfile = () => {
+    let newErrors: Errors = {};
+
+    if (!editedData.user_name?.trim()) {
+      newErrors.user_name = "Username is required";
+    }
+    if (!editedData.display_name?.trim()) {
+      newErrors.display_name = "Display Name is required";
+    }
+    if (!editedData.email?.trim()) {
+      newErrors.Email = "Email is required";
+    } else if (!validateEmail(editedData.email)) {
+      newErrors.Email = "Invalid email address";
+    }
+    setErrors(newErrors);
+
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const editUser = async (id: number, updatedData: RowData) => {
```

Comment:
Give a relevant name for the function

Reply:
relevant name for function has been changed